webapp/components/shortcuts_modal.jsx

```diff
@@ -0,0 +1,392 @@
+    browserHighlightNext: {
+        id: 'shortcuts.browser.highlight_next',
+        defaultMessage: 'Highlight text to the next line:\tShift|Down'
+    },
+    browserNewline: {
+        id: 'shortcuts.browser.newline',
+        defaultMessage: 'Create a new line:\tShift|Enter'
+    },
+    info: {
+        id: 'shortcuts.info',
+        defaultMessage: 'Begin a message with / for a list of all the commands at your disposal.'
+    }
+});
+
+class ShortcutsModal extends React.PureComponent {
+    static propTypes = {
+        intl: intlShape.isRequired
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            show: false
+        };
+    }
+
+    componentDidMount() {
+        ModalStore.addModalListener(Constants.ActionTypes.TOGGLE_SHORTCUTS_MODAL, this.handleToggle);
+    }
+
+    componentWillUnmount() {
+        ModalStore.removeModalListener(Constants.ActionTypes.TOGGLE_SHORTCUTS_MODAL, this.handleToggle);
+    }
+
+    handleToggle = (value) => {
+        this.setState({
+            show: value
+        });
+    }
+
+    handleHide = () => {
+        this.setState({show: false});
+    }
+
+    render() {
+        let navPrev = shortcuts.navPrev;
+        let navNext = shortcuts.navNext;
+        let navUnreadPrev = shortcuts.navUnreadPrev;
+        let navUnreadNext = shortcuts.navUnreadNext;
+        let navSwitcher = shortcuts.navSwitcher;
+        let navSwitcherTeam = shortcuts.navSwitcherTeam;
+        let navDMMenu = shortcuts.navDMMenu;
+        let navSettings = shortcuts.navSettings;
+        let navMentions = shortcuts.navMentions;
+        let msgReprintPrev = shortcuts.msgReprintPrev;
+        let msgReprintNext = shortcuts.msgReprintNext;
+        let filesUpload = shortcuts.filesUpload;
+        let browserChannelPrev = shortcuts.browserChannelPrev;
+        let browserChannelNext = shortcuts.browserChannelNext;
+        let browserFontIncrease = shortcuts.browserFontIncrease;
+        let browserFontDecrease = shortcuts.browserFontDecrease;
+        if (Utils.isMac()) {
+            navPrev = shortcuts.navPrevMac;
+            navNext = shortcuts.navNextMac;
+            navUnreadPrev = shortcuts.navUnreadPrevMac;
+            navUnreadNext = shortcuts.navUnreadNextMac;
+            navSwitcher = shortcuts.navSwitcherMac;
+            navSwitcherTeam = shortcuts.navSwitcherTeamMac;
+            navDMMenu = shortcuts.navDMMenuMac;
+            navSettings = shortcuts.navSettingsMac;
+            navMentions = shortcuts.navMentionsMac;
+            msgReprintPrev = shortcuts.msgReprintPrevMac;
+            msgReprintNext = shortcuts.msgReprintNextMac;
+            filesUpload = shortcuts.filesUploadMac;
+            browserChannelPrev = shortcuts.browserChannelPrevMac;
+            browserChannelNext = shortcuts.browserChannelNextMac;
+            browserFontIncrease = shortcuts.browserFontIncreaseMac;
+            browserFontDecrease = shortcuts.browserFontDecreaseMac;
+        }
```

Comment:
Could you separate out the shortcuts by OS and replace this with something like...

```js
let shortcuts = defaultShortcuts;
if (Utils.isMac()) {
    osShortcuts = macShortcuts;
}
```

?

Just not a fan of lists like this. I think they're relatively error-prone and easy to avoid.